### ISSUE TITLE
AASTeX format for software records

### ISSUE THREAD

**Issue by ehenneken** [adsabs/export_service#81]
If the 'doctype' of a record is `software`, the AASTeX export of a record should be formatted as follows:
```
\bibitem[...]{bibcode}  {authors} {year}, {title}, {version}, {publisher}, doi:{doi}
```
(for software records with a DOI) where `version` and `publisher` are taken from the Solr data or left blank if not populated. The current output for ASCL records can remain unchanged. For example
```
\bibitem[Janiuk(2018)]{2018ascl.soft12002J} Janiuk, A.\ 2018, Astrophysics Source Code Library , ascl:1812.002.
```

Zenodo example: the AASTeX export for `2016zndo.soft53155F` should be
```
\bibitem[Foreman-Mackey et al.(2016)]{2016zndo.soft53155F} Foreman-Mackey, D., Vousden, W., Price-Whelan, A., et al.\ 2016, corner.py: corner.py v2.0.0, v2.0.0, Zenodo, doi:10.5281/zenodo.53155
```
The fact that `v2.0.0` occurs twice is just an unfortunate side-effect of metadata that was created automatically. There are plenty of Zenodo software records where the title of the record is really just a title, so examples like the one above should be viewed as aberrations. For now, we shouldn't do anything extra to clean up these aberrations.

**Comment by golnazads** [comment]
when you say `ASCL records can remain unchanged` if there is version or publisher associated with ascl record and we are outputting those fields for AASTex, then the format is not going to stay the same. so my question is for ascl record, are you going to have version and publisher in solr? As I mentioned on slack, solr is going to dictate the fields, if they are in solr, they are going to get displayed.

**Comment by golnazads** [comment]
sorry, I have to take it back. we do distinguish between the doctype, but it is at the level of CSL
```
  <macro name="container">
    <choose>
      <if type="chapter paper-conference" match="any">
        <names variable="editor translator" delimiter=", " prefix=" " suffix=", ">
          <name delimiter-precedes-last="always" initialize-with="." name-as-sort-order="all"/>
          <label form="short" text-case="capitalize-first" prefix=" (" suffix=")"/>
        </names>
        <group delimiter=", ">
          <text variable="container-title" text-case="title"/>
          <text variable="collection-title" text-case="title"/>
        </group>
      </if>
      <else-if type="book report" match="any">
        <group delimiter=", ">
          <text variable="container-title"/>
          <text variable="collection-title"/>
        </group>
      </else-if>
      <else-if type="article" match="any">
        <group delimiter=", ">
          <text variable="container-title"/>
          <text variable="collection-title"/>
        </group>
      </else-if>
      <else>
        <group delimiter=", ">
          <text variable="container-title"/>
          <text variable="collection-title"/>
        </group>
      </else>
    </choose>
  </macro>
```

**Comment by golnazads** [comment]
Beside deciding if you want to have page be displayed for software, need to also decide on title/pub field. Previously we were displaying `pub` field for ascl, now since they are marked as software, if we output title then we will have
```
\bibitem[Janiuk(2018)]{2018ascl.soft12002J} Janiuk, A.\ 2018, GLADIS: GLobal Accretion Disk Instability Simulation.
```
instead of what we had
```
\bibitem[Janiuk(2018)]{2018ascl.soft12002J} Janiuk, A.\ 2018, Astrophysics Source Code Library , ascl:1812.002.
```
from solr
```
      {
        "bibcode":"2018ascl.soft12002J",
        "pub":"Astrophysics Source Code Library",
        "title":["GLADIS: GLobal Accretion Disk Instability Simulation"]}
```

**Comment by golnazads** [comment]
Updated the code for software as specified and for the two records with version and publisher we have
```
{
    "msg": "Retrieved 2 abstracts, starting with number 1.",
    "export": "\\bibitem[Foreman-Mackey et al.(2016)]{2016zndo.soft53155F} Foreman-Mackey, D., Vousden, W., Price-Whelan, A., et al.\\ 2016, corner.py: corner.py v2.0.0, v2.0.0, Zenodo, doi:10.5281/zenodo.53155.\n\\bibitem[Foreman-Mackey et al.(2016)]{2016zndo.soft45906F} Foreman-Mackey, D., Vousden, W., Price-Whelan, A., et al.\\ 2016, corner.py: corner.py v1.0.2, v1.0.2, Zenodo, doi:10.5281/zenodo.45906.\n"
}
```

**Comment by golnazads** [comment]
implemented in v2.0.35
